src/main/java/com/hypherionmc/modpublisher/plugin/ModPublisherGradleExtension.java

```diff
@@ -555,6 +554,69 @@ public class GithubConfig {
          */
         @ApiStatus.Experimental
         private boolean updateRelease = true;
+
+        /**
+         * Kotlin Compatibility setter
+         * Set GitHub Release tag. Defaults to Version
+         * @param tag The tag to target
+         */
+        public void tag(String tag) {
+            this.tag = tag;
+        }
```

Comment:
Do you mind explaining why this is needed? My understanding was that kotlin would convert `setXyz` to `xyz =` just fine :thinking:

Reply:
This is what happens when you try to use 

```kotlin
github {
        tag = "v${project.version}"
        createTag = true
        createRelease = true
        updateRelease = true
    }
```
    
    without the manual setters:
    
```
e: /run/media/hypherionsa/Data/coding/modpublisher/testprojects/kotlintest/build.gradle.kts:66:9: Val cannot be reassigned
e: /run/media/hypherionsa/Data/coding/modpublisher/testprojects/kotlintest/build.gradle.kts:67:9: Cannot access 'createTag': it is private in 'GithubConfig'
e: /run/media/hypherionsa/Data/coding/modpublisher/testprojects/kotlintest/build.gradle.kts:68:9: Cannot access 'createRelease': it is private in 'GithubConfig'
e: /run/media/hypherionsa/Data/coding/modpublisher/testprojects/kotlintest/build.gradle.kts:69:9: Cannot access 'updateRelease': it is private in 'GithubConfig'

Build file '/run/media/hypherionsa/Data/coding/modpublisher/testprojects/kotlintest/build.gradle.kts' line: 66
```

Kotlin would expect setTag, setCreateTag etc, when using the lombok setters

Reply:
I guess I misremembered:
![image](https://github.com/firstdarkdev/modpublisher/assets/5046562/4d6a72e2-7532-487b-9fa6-6d89cf97e573)

I guess this only applies to `var` properties written in kotlin?

Reply:
Could be yeah.

I'll be completely honest, my knowledge of Kotlin based buildscripts, and even kotlin itself is very limited. I understand kotlin well enough to figure out what, where and how, but I haven't written anything in Kotlin myself.

So my initial way of doing it is probably also the wrong way of doing it, or not even ideal, but it's one that works :shrug: 

I do hope to one day improve my kotlin knowledge and redo this properly from the start

Reply:
As I said in the [other thread](https://github.com/firstdarkdev/modpublisher/pull/11#discussion_r1550302699), I agree this should be consistent with the existing setters.

I've not used kotlin that much either. I'm sure this could be improved in a future breaking change and/or kotlin port :rocket: